### PR TITLE
=act Fix dns query overriding in dnsClient.

### PR DESCRIPTION
**Pull request by He-Pin**
Refs https://github.com/akka/akka/issues/31905

If there are multi queries issued with the same id, the current DNS client may reply the result to the wrong target.

### CODE REVIEW

File: akka-actor/src/main/scala/akka/io/dns/internal/DnsClient.scala [hunk 1]
```diff
@@ -111,6 +106,7 @@ import akka.pattern.{ BackoffOpts, BackoffSupervisor }
               case (s, _) =>
                 s ! Failure(new RuntimeException("Send failed to nameserver"))
                 inflightRequests -= msg.id
+                unstashQuestion(msg.id)
```
Comment (leviramsey) [inline akka-actor/src/main/scala/akka/io/dns/internal/DnsClient.scala:109]: Could just be `unstashAll`... the questions that still collide will still collide and get re-stashed.

Comment (He-Pin) [reply akka-actor/src/main/scala/akka/io/dns/internal/DnsClient.scala:109]: That's true

**Comment by johanandren** [comment]
Sorted by #31906

**Comment by He-Pin** [comment]
i think this cover different part. but feel free.

**Comment by johanandren** [comment]
Ah, sorry, yes, this is not the async dns, but the "regular" one.

**Comment by He-Pin** [comment]
If this is valid I may update this pr during the holiday.

**Pull request closed without merging**